packages/@headlessui-react/src/test-utils/interactions.ts

```diff
@@ -486,6 +488,14 @@ let focusableSelector = [
   )
   .join(',')
 
+let focusableSelectorWithNegativeTabindex = _focusableSelector
+  .map(
+    process.env.NODE_ENV === 'test'
+      ? (selector) => `${selector}:not([style*='display: none'])`
+      : (selector) => selector
+  )
+  .join(',')
```

Comment:
An element with `tabIndex=-1` can be focused on through means other than keyboard navigation, such as clicking.